symengine/sets.h

```diff
@@ -0,0 +1,62 @@
+/**
+ *  \file sets.h
+ *  Polynomial Manipulation
+ *
+ **/
+#ifndef SYMENGINE_SETS_H
+#define SYMENGINE_SETS_H
+
+#include <symengine/basic.h>
+#include <symengine/functions.h>
+#include <symengine/number.h>
+#include <symengine/rational.h>
+#include <symengine/complex.h>
+
+
+namespace SymEngine {
+
+class Interval : public Basic  {
```

Comment:
I thought over it while implementing.
But as sympy uses -
`class Set(Basic)`
And in `Number` we have virtual functions like -
`nth_root`, `is_perfect_power`, `pow` etc which don't make sense for a set object.
So, I thought using `Basic` will be better.